tests/unittest_tests.bzl

```diff
@@ -177,6 +177,63 @@ inspect_actions_test = analysistest.make(
     _inspect_actions_test,
 )
 
+########################################
+####### inspect_output_dirs_test #######
+########################################
+_OutputDirInfo = provider(fields = ["bin_path", "genfiles_path"])
+
+def _inspect_output_dirs_test(ctx):
+    """Test verifying output directories used by a test."""
+    env = analysistest.begin(ctx)
+
+    # Assert that the output dirs observed by the aspect added by analysistest
+    # are the same as those observed by the rule directly, even when that's
+    # under a config transition and therefore not the same as the output
+    # dirs used by the test rule.
+    bin_path = analysistest.target_bin_dir_path(env)
+    genfiles_path = analysistest.target_genfiles_dir_path(env)
+    target_under_test = analysistest.target_under_test(env)
+    asserts.false(env, not bin_path, "bin dir path not found.")
+    asserts.false(env, not genfiles_path, "genfiles path not found.")
+    asserts.false(
+        env,
+        bin_path == ctx.bin_dir.path,
+        "bin dir path expected to differ between test and target_under_test.",
+    )
+    asserts.false(
+        env,
+        genfiles_path == ctx.genfiles_dir.path,
+        "genfiles dir path expected to differ between test and target_under_test.",
+    )
+    asserts.equals(env, bin_path, target_under_test[_OutputDirInfo].bin_path)
+    asserts.equals(
+        env,
+        genfiles_path,
+        target_under_test[_OutputDirInfo].genfiles_path,
+    )
+    return analysistest.end(env)
+
+def _inspect_output_dirs_fake_rule(ctx):
+    return [
+        _OutputDirInfo(
+            bin_path = ctx.bin_dir.path,
+            genfiles_path = ctx.genfiles_dir.path,
+        ),
+    ]
+
+inspect_output_dirs_fake_rule = rule(
+    implementation = _inspect_output_dirs_fake_rule,
+)
+
+inspect_output_dirs_test = analysistest.make(
+    _inspect_output_dirs_test,
+    # The output directories differ between the test and target under test when
+    # the target under test is under a config transition.
+    config_settings = {
+        "//command_line_option:compilation_mode": "fastbuild",
```

Comment:
Does this mean that these tests will now fail if tested under fastbuild?
That is, if I invoke:

```
bazel build tests:all -c=fastbuild
```

I will see test failures?

We should try to avoid that, and select at least an example option that is unlikely to be set.
How about instead:

```
"//command_line_option:test_arg": ["constant_nobody_will_use"],
```

Reply:
That doesn't seem to be the case (at least not downstream). The targets under the config transition have different output dirs even if it ends up being a no-op.

Reply:
Ah, interesting. Thanks.